server/index.js

```diff
@@ -38,6 +38,29 @@ app.get('/api/products', (req, res, next) => {
     });
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+
+  if (!parseInt(productId)) {
+    next(new ClientError('Product id must be a positive integer.', 400));
+  }
+
+  const sql = `
+    select *
+      from "products"
+      where "productId" = $1;
+    `;
+  const params = [productId];
+
+  db.query(sql, params)
+    .then(result => {
+      res.status(200).json(result.rows);
```

Comment:
Last thing here. Be sure to send the matching product (just the object), instead of an array containing the object.